Adding support to Custom Aggregation Methods

athoscouto opened the pull request:
### Description
*Enhancing AggregationMethod class and modifying UriQueryExpressionParser to support Custom Aggregation Methods according to OData 4.0 spec. Right now only methods that return doubles are supported*

### Checklist (Uncheck if it is not completed)
- [x] Test cases added
- [x] VS Build and tests passed

### Observations 
*We need to find out the best way to address the TODOs in the code. In the return type of the aggregation methods: right now we are limited to methods that returns doubles, because the parses don't have information about the functions registered.*

kosinsky commented on src/Microsoft.OData.Core/UriParser/Aggregation/AggregationMethod.cs:
```diff
@@ -9,7 +9,31 @@ namespace Microsoft.OData.UriParser.Aggregation
+            this.MethodType = aggregationMethodType;
+        }
+
+        public static AggregationMethod Sum = new AggregationMethod(AggregationMethodType.Sum);
```
Could you add readonly?

athoscouto replied:
Tried to, got some errors about the reference type being mutable. As we talked offline, won't do it.

robward-ms committed:
Issues
This pull request fixes issue #706.

Description
Consider owning type when adding/getting properties from the PropertyCacheHandler.
Unit tests are added to cover simple PropertyCacheHandler scenarios.

Checklist (Uncheck if it is not completed)
[ x ] Test cases added
[ x ] Build and test with one-click build and test script passed

Additional work necessary
No document change required.

mikepizzo committed:
Merge pull request #756 from robward-ms/property-cache

shinlges committed:
Adding support and tests for virtual property count.

shinlges committed:
Removing compiler warnings, adding negative test case, removing redundant code and adding comments referencing issue #758.

shinlges committed:
Reordering comments to avoid compiler warnings.

robward-ms committed:
Update nuget.exe to 3.5.0 to resolve build issue with xunit.

shinlges committed:
Adding support and tests for custom aggregation methods.
